test/Groups/constructors.jl

```diff
@@ -65,6 +65,10 @@ end
   @test isa(dihedral_group(PermGroup, 6), PermGroup)
 
   @test is_quaternion_group(small_group(8, 4))
+  @test ! is_quaternion_group(small_group(12, 3))
+  @test is_dicyclic_group(small_group(8, 4))
+  @test ! is_dicyclic_group(small_group(13, 1))
```

Comment:
A test where `is_dicyclic_group(G)` is `true`  but `is_quaternion_group(G)` is `false` would be useful, for example
```
G = small_group(12, 1);
@test is_dicyclic_group(G) && !is_quaternion_group(G)
```
(This was the motivation for adding `IsDicyclicGroup` to GAP.)